File: python/sorting_algorithms/bubble_sort.py
from sorting_algorithm import SortingAlgorithm as SA

class BubbleSort(SA):
	def sort(self, a_list):
		exch = True

		while (exch):
			exch = False
			for i in range(len(a_list) - 1):
				if a_list[i] > a_list[i+1]:
					exch = True
					a_list[i], a_list[i+1] = a_list[i+1], a_list[i]

File: java/sorting_algorithms/BubbleSort.java
import java.util.ArrayList;

public class BubbleSort extends SortClass {
	public ArrayList<Double> sort(ArrayList<Double> al) {
		boolean exch = true;

		while (exch) {
			exch = false;

			for (int i = 1; i < al.size(); i++) {
				if (Double.compare(al.get(i), al.get(i-1)) < 0) {
					exch = true;
					Double temp = al.get(i);
					al.set(i, al.get(i-1));
					al.set(i-1, temp);
				}
			}
		}

		return al;
	}
}